raiden_libs/network/matrix/client.py

```diff
@@ -22,9 +23,38 @@ class GMatrixHttpApi(MatrixHttpApi):
     of concurrent requests we make to the number of connections
     available to us in requests.Session connection pool size.
     """
-    def __init__(self, *args, **kwargs):
-        super().__init__(*args, **kwargs)
-        self.lock = Semaphore(DEFAULT_POOLSIZE)
+    def __init__(
+            self,
+            base_url,
+            token=None,
+            identity=None,
+            default_429_wait_ms=5000,
```

Comment:
I think it'd be better to have above parameters just fall under `*args` and `**kwargs` and spread it to super init. If we're not using them in our constructor, no need to force us keeping in-sync with the interface here.

Reply:
`max_retries` and `poll_maxsize`is not part of the super interface. either we had to remove them or be forced to refer the arguments from super by name or position only

Reply:
You can just put them before `*args, **kwargs`, and just be sure to pass them first or by name, it should be enough to 'extract' them from the list of the superclass parameters.

Reply:
adding them before will break the constructor interface

Reply:
Please review the change.